### PR TITLE
Attack feature

### PR DESCRIPTION
**Pull request by NickWilsonDev**
Please review

### CODE REVIEW

**Comment by NickWilsonDev** [comment]
@nybblr please review

**Comment by nybblr** [comment]
Taking a peak now!

File: SecretOfManaClone.py [hunk 1]
```diff
@@ -1,19 +1,123 @@
...
+        images.append(pygame.image.load("./art/dk_down_attack1.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_down_attack2.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_down_attack3.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_down_attack4.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_down_attack5.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_down_attack6.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_down_attack7.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_down_attack8.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_down_attack9.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_down_attack10.png").convert_alpha())
```
Comment (nybblr) [inline SecretOfManaClone.py]: That's a lot of copy pasted lines, could we use a loop for this?

File: SecretOfManaClone.py [hunk 2]
```diff
@@ -1,19 +1,123 @@
...
+        images.append(pygame.image.load("./art/dk_up_attack1.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_up_attack2.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_up_attack3.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_up_attack4.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_up_attack5.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_up_attack6.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_up_attack7.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_up_attack8.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_up_attack9.png").convert_alpha())
+        images.append(pygame.image.load("./art/dk_up_attack10.png").convert_alpha())
```
Comment (nybblr) [inline SecretOfManaClone.py]: ^^ same question

File: SecretOfManaClone.py [hunk 3]
```diff
@@ -1,19 +1,123 @@
...
+
+def get_standing_direction(direction):
+    if direction == "up":
+        image = pygame.image.load("./art/dk_up.png").convert_alpha()
+    elif direction == "down":
+        image = pygame.image.load("./art/dk_down.png").convert_alpha()
+    elif direction == "right":
+        image = pygame.image.load("./art/dk_right.png").convert_alpha()
+    elif direction == "left":
+        image = pygame.image.load("./art/dk_left.png").convert_alpha()
```
Comment (nybblr) [inline SecretOfManaClone.py]: Ooh, I wonder if you could just use string interpolation to grab the image based on the direction?

```python
return pygame.image.load("./art/dk_%s.png" % direction).convert_alpha()
```

File: SecretOfManaClone.py [hunk 4]
```diff
@@ -1,19 +1,123 @@
...
+    return image
+
+def display_menu():
+    # list = cycle(images) to cycle through circular list
+    images = []
+    images.append(pygame.image.load("./art/undine.png").convert_alpha())
+    images.append(pygame.image.load("./art/shade.png").convert_alpha())
+    images.append(pygame.image.load("./art/sylphid.png").convert_alpha())
+    images.append(pygame.image.load("./art/salamando.png").convert_alpha())
```
Comment (nybblr) [inline SecretOfManaClone.py]: A list comprehension could help! e.g. have a list of the file names:

```python
images = ['undine', 'shade', 'sylphid', 'salamando']
# loop over the list!
```

File: SecretOfManaClone.py [hunk 5]
```diff
@@ -1,19 +1,123 @@
...
+    return images
 
 def main():
-    width = 1500
-    height = 900
+    screen_width = 1500
+    screen_height = 900
```
Comment (nybblr) [inline SecretOfManaClone.py:56]: :+1: This definitely helps with readability!

File: SecretOfManaClone.py [hunk 6]
```diff
@@ -59,24 +171,33 @@ def main():
...
+        elif keys[pygame.K_LEFT] and not item_menu:
             character_x = character_x - 5
             character_facing = "left"
+            character_walk = True
 
-        elif keys[pygame.K_RIGHT]:
+        elif keys[pygame.K_RIGHT] and not item_menu:
             character_x = character_x + 5
             character_facing = "right"
+            character_walk = True
```
Comment (nybblr) [inline SecretOfManaClone.py:131]: This seems like it could be refactored somehow, ideas?

File: SecretOfManaClone.py (hunk 1 above)
Comment (NickWilsonDev) [reply SecretOfManaClone.py]: Thanks, I was able to get rid of quite a few lines of code with these suggestions.

**Pull request merged**